docs/source/building/variants.rst

```diff
@@ -0,0 +1,1230 @@
+
+
+and conda-build would loop over these variants where appropriate (building,
+outputting package output names, etc.)
+
+If ``numpy`` had had two values instead of one, we'd end up with *four* output
+variants: 2 variants for ``python``, *times* two variants for ``numpy``:
+
+.. code-block:: python
+
+    variants = [{'python': '3.4', 'numpy': '1.11'}, {'python': '3.5', 'numpy': '1.11'},
+                {'python': '3.4', 'numpy': '1.10'}, {'python': '3.5', 'numpy': '1.10'}]
+
+
+Bootstrapping pins based on an existing environment
+---------------------------------------------------
+
+
+To establish your initial variant, you may point at an existing conda
+environment. Conda-build will examine the contents of that environment and pin
+to the exact requirements that make up that environment.
+
+.. code-block:: shell
+
+   conda build --bootstrap name_of_env
+
+
+You may specify either environment name (and depend on conda's environment
+lookup) or filesystem path to the environment.
+
+
+Extended keys
+-------------
+
+
+These are not looped over to establish the build matrix. Rather, they are
+aggregated from all input variants, and each derived variant shares the whole
+set. These are used internally for tracking which requirements should be pinned,
+for example, with the ``pin_run_as_build`` key. You can add your own extended
+keys by passing in values for the ``extend_keys`` key for any variant.
+
+For example, if you wanted to collect some aggregate trait from multiple
+conda_build_config.yaml files, you could do something like this:
+
+HOME/conda_build_config.yaml:
+
+.. code-block:: yaml
+
+   some_trait:
+     - dog
+   extend_keys:
+     - some_trait
+
+
+recipe/conda_build_config.yaml:
+
+.. code-block:: yaml
+
+   some_trait:
+     - pony
+   extend_keys:
+     - some_trait
+
+Note that *both* of the conda_build_config.yaml files need to list the trait as
+an ``extend_keys`` entry.  If you list it in only one of them, an error will be
+raised, to avoid confusion with one conda_build_config.yaml file that would add
+entries to the build matrix, and another which would not. For example, this
+should raise an error:
+
+.. code-block:: yaml
+
+   some_trait:
+     - dog
+
+
+recipe/conda_build_config.yaml:
+
+.. code-block:: yaml
+
+   some_trait:
+     - pony
+   extend_keys:
+     - some_trait
+
+
+When our two proper yaml config files are combined, ordinarily the recipe-local
+variant would clobber the user-wide variant, yielding ``{'some_trait':
+'pony'}``. However, with the extend_keys entry, we end up with what we've always
+wanted: a dog *and* pony show: ``{'some_trait': ['dog', 'pony'])}``
+
+Again, mostly an internal implementation detail - unless you find a use for it.
+Internally, it is used to aggregate the ``pin_run_as_build`` and
+``ignore_version`` entries from any of your conda_build_config.yaml
+files.
+
+
+Customizing compatibility
+-------------------------
+
+.. _pinning_expressions:
+
+Pinning expressions
```

Comment:
@msarahan The code in the first example only seems to mention 1.11 but the comment mentions 1.11.2. Is this correct?